client/components/TwoFactorModal/TwoFactorTotp.tsx

```diff
@@ -0,0 +1,51 @@
+import { Box, TextInput, Icon } from '@rocket.chat/fuselage';
+import { useAutoFocus } from '@rocket.chat/fuselage-hooks';
+import React, { ReactElement, useState, ChangeEvent } from 'react';
+
+import { useTranslation } from '../../contexts/TranslationContext';
+import GenericModal from '../GenericModal';
+import { Method, OnConfirm } from './TwoFactorModal';
+
+type TwoFactorTotpProps = {
+	onConfirm: OnConfirm;
+	onClose: () => void;
+};
+
+const TwoFactorTotp = ({ onConfirm, onClose }: TwoFactorTotpProps): ReactElement => {
+	const t = useTranslation();
+	const [code, setCode] = useState<string>('');
+	const ref = useAutoFocus();
+
+	const onConfirmTotpCode = (): void => {
+		onConfirm(code, Method.TOTP);
+	};
+
+	const onChange = ({ currentTarget }: ChangeEvent<HTMLInputElement>): void => {
+		setCode(currentTarget.value);
+	};
+
+	return (
+		<GenericModal
+			onConfirm={onConfirmTotpCode}
+			onCancel={onClose}
```

Comment:
`onCancel` is dispatching unnecessary error

Reply:
As we talked, the error is not unecessary, as it depicts the action wasn't succesfull if TwoFactor is canceled.